core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,277 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for solution.
+ */
+
+oppia.controller('StateSolution', [
+  '$scope', '$rootScope', '$modal', '$filter', '$injector',
+  'editorContextService', 'alertsService', 'INTERACTION_SPECS',
+  'stateSolutionService', 'explorationStatesService',
+  'StateSolutionHelperService', 'oppiaExplorationHtmlFormatterService',
+  'stateInteractionIdService', 'stateHintsService', 'UrlInterpolationService',
+  'SolutionObjectFactory', 'responsesService', 'AnswerClassificationService',
+  'explorationContextService', 'angularNameService',
+  function(
+    $scope, $rootScope, $modal, $filter, $injector,
+    editorContextService, alertsService, INTERACTION_SPECS,
+    stateSolutionService, explorationStatesService,
+    StateSolutionHelperService, oppiaExplorationHtmlFormatterService,
+    stateInteractionIdService, stateHintsService, UrlInterpolationService,
+    SolutionObjectFactory, responsesService, AnswerClassificationService,
+    explorationContextService, angularNameService) {
+    $scope.editorContextService = editorContextService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.isActive = false;
+    $scope.answerCheckPassed = false;
+    $scope.INTERACTION_SPECS = INTERACTION_SPECS;
+    $scope.interactionHtml = '';
+
+    $scope.stateHintsService = stateHintsService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.correctAnswer = null;
+
+    $scope.submitAnswer = function (answer) {
+      $scope.correctAnswer = answer;
+    };
+
+    $scope.toggleIsActive = function() {
+      $scope.isActive = !$scope.isActive;
+    };
+
+    $scope.$on('stateEditorInitialized', function(evt, stateData) {
+      stateSolutionService.init(
+        editorContextService.getActiveStateName(),
+        stateData.interaction.solution);
+      $scope.interactionHtml = (
+        oppiaExplorationHtmlFormatterService.getInteractionHtml(
+          stateInteractionIdService.savedMemento,
+          explorationStatesService.getInteractionCustomizationArgsMemento(
+            editorContextService.getActiveStateName()),
+          'currentInteractionHtml'));
+
+      $scope.solutionIsSpecified = (
+        !!JSON.stringify(stateSolutionService.savedMemento));
+      $scope.isActive = false;
+      var interactionId = stateInteractionIdService.savedMemento;
+
+      $scope.INTERACTION_SPECS = INTERACTION_SPECS[interactionId];
+
+      if (StateSolutionHelperService.isSupportedInteraction(interactionId)) {
+        $scope.interactionHtml = null;
+        $scope.objectType = (
+          StateSolutionHelperService.getUnsupportedInteractionObjectType(
+            interactionId));
+      } else {
+        $scope.objectType = null;
+      }
+    });
+
+    $scope.getSolutionSummary = function() {
```

Comment:
This seems better defined as a method on the Solution object. It needs Karma tests.